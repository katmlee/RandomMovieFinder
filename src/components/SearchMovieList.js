import React, { PropTypes } from 'react';
import MovieList from './MovieList';

const SearchMovieList = ({ navigation }) => (
  <MovieList movies={navigation.state.params.movies} navigation={navigation} />
);

SearchMovieList.navigationOptions = ({ navigation }) => ({
  title: `Searches for ${navigation.state.params.keyword}`,
});

SearchMovieList.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func,
  }).isRequired,
};

export default SearchMovieList;
