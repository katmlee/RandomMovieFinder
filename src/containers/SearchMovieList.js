import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { ActivityIndicator, Text, View } from 'react-native';
import MovieList from '../components/MovieList';

const SearchMovieList = (props) => {
  const { navigation, isLoading, movies, error } = props;
  if (isLoading) {
    return <ActivityIndicator color={'red'} style={{ flex: 1 }} />;
  }

  if (error) {
    return (
      <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
        <Text>Something went wrong</Text>
      </View>
    );
  }

  return <MovieList movies={movies} navigation={navigation} />;
};

SearchMovieList.navigationOptions = ({ navigation }) => ({
  title: `Searches for ${navigation.state.params.keyword}`,
});

SearchMovieList.defaultProps = {
  movies: null,
  error: null,
  isLoading: false,
};

SearchMovieList.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func,
  }).isRequired,
  movies: PropTypes.arrayOf(PropTypes.shape({})),
  error: PropTypes.shape({}),
  isLoading: PropTypes.bool,
};

const mapStateToProps = state => ({
  movies: state.movies.movies,
  error: state.movies.error,
  isLoading: state.movies.isLoading,
});

export default connect(mapStateToProps)(SearchMovieList);
