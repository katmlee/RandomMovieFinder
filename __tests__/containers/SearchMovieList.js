import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import SearchMovieList from '../../src/containers/SearchMovieList';

const mockStore = configureStore();
let store;
const navigation = {
  navigate: jest.fn(),
};

describe('SearchMovieList', () => {
  it('should show movieList', () => {
    const movies = {
      movies: [],
    };
    store = mockStore({ movies });
    const tree = renderer.create(
      <SearchMovieList store={store} navigation={navigation} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should show activity indicator if loading', () => {
    const movies = {
      movies: [],
      isLoading: true,
    };
    store = mockStore({ movies });
    const tree = renderer.create(
      <SearchMovieList store={store} navigation={navigation} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should show error if there is error', () => {
    const movies = {
      movies: [],
      error: {},
    };
    store = mockStore({ movies });
    const tree = renderer.create(
      <SearchMovieList store={store} navigation={navigation} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
