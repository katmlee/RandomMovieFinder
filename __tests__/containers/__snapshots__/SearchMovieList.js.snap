// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchMovieList should show activity indicator if loading 1`] = `
<ActivityIndicator
  animating={true}
  color="red"
  hidesWhenStopped={true}
  size="small"
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;

exports[`SearchMovieList should show error if there is error 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
    Something went wrong
  </Text>
</View>
`;

exports[`SearchMovieList should show movieList 1`] = `
<RCTScrollView
  data={Array []}
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  onViewableItemsChanged={undefined}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "flex": 1,
      "width": "100%",
    }
  }
  updateCellsBatchingPeriod={50}
  windowSize={21}
>
  <View />
</RCTScrollView>
`;
